torch/functional.py

```diff
@@ -262,76 +262,100 @@ def lu_unpack(LU_data, LU_pivots, unpack_data=True, unpack_pivots=True):
-
-
-    >>> A = torch.randn(3,5,4)
-    >>> l = torch.randn(2,5)
-    >>> r = torch.randn(2,4)
-    >>> torch.einsum('bn,anm,bm->ba', l, A, r) # compare torch.nn.functional.bilinear
-    tensor([[-0.3430, -5.2405,  0.4494],
-            [ 0.3311,  5.5201, -3.0356]])
-
-
-    >>> As = torch.randn(3,2,5)
-    >>> Bs = torch.randn(3,5,4)
-    >>> torch.einsum('bij,bjk->bik', As, Bs) # batch matrix multiplication
-    tensor([[[-1.0564, -1.5904,  3.2023,  3.1271],
-             [-1.6706, -0.8097, -0.8025, -2.1183]],
-
-            [[ 4.2239,  0.3107, -0.5756, -0.2354],
-             [-1.4558, -0.3460,  1.5087, -0.8530]],
-
-            [[ 2.8153,  1.8787, -4.3839, -1.2112],
-             [ 0.3728, -2.1131,  0.0921,  0.8305]]])
-
-    >>> A = torch.randn(3, 3)
-    >>> torch.einsum('ii->i', A) # diagonal
-    tensor([-0.7825,  0.8291, -0.1936])
-
-    >>> A = torch.randn(4, 3, 3)
-    >>> torch.einsum('...ii->...i', A) # batch diagonal
-    tensor([[-1.0864,  0.7292,  0.0569],
-            [-0.9725, -1.0270,  0.6493],
-            [ 0.5832, -1.1716, -1.5084],
-            [ 0.4041, -1.1690,  0.8570]])
-
-    >>> A = torch.randn(2, 3, 4, 5)
-    >>> torch.einsum('...ij->...ji', A).shape # batch permute
-    torch.Size([2, 3, 5, 4])
-"""
+    Computes multilinear expressions (i.e. sums of products) specified using the Einstein summation convention.
+
+    The Einstein summation convention provides a succint way of specifying many linear algebra operations by
+    labeling each dimension of the input operands and specifying which dimensions are to be summed over.
+    Einsum provides two modes: implicit mode computes the classical Einstein summation while explicit mode
+    provides more flexibility by manually enabling/disabling summation. The following section describes the
+    formatting and rules for the equation.
+
+    Equation:
+
+        The :attr:`equation` string specifies the subscripts (lower case letters `['a', 'z']`) for the Eintein summation.
+
+        In implicit mode, :attr:`equation` specifies the subscripts for each dimension of the input :attr:`operands` in
+        order, separating subcripts for each operand by a comma (','), e.g. `'ij,jk'` specify subscripts for two 2D
+        operands. The dimensions labeled with the same subscript must be broadcastable, that is, their size must either
+        match or be `1`. The exception is if a subscript is repeated for the same input operand, in which case the
+        dimensions under the subscript for the operand must match in size and the diagonal of the corresponding dimensions
+        will be taken. The dimensions that appear exactly once in the :attr:`equation` will be part of the output Tensor,
+        sorted in increasing alphabetical order. The dimensions that are not part of the output (appear more than once)
+        will be summed over.
+
+        In explicit mode, the same rules for the input operands in implicit mode apply but the output dimensions can be
```

Comment:
Is it just that the dimensions can be reordered in the output? How should a user understand the "default" ordering/naming of output dimensions?